### PR TITLE
Updating Analytics Tracking

### PR DESCRIPTION
**Pull request by mendelB**
#### What's this PR do?
This PR updates and expands the analytics tracking events we have in place for our various forms.

- 🗑  https://github.com/DoSomething/northstar/commit/6606b14426ce8f884f2e181ff9b86401470d54f7 Removes the separate `Puck.js` file in favor of having all our analytics in the `Analytics.js` file so we can control things more easily and achieve better parity between our Puck and GA tracking.
- 👯‍♂️ 👯  https://github.com/DoSomething/northstar/commit/cf7f6235fa3c93a959f6b73305cbed929df2a34f Add the necessary Puck and GA events so that we're tracking things more in parallel between the two platforms. (Left out a couple which we deemed unnecessary for Puck)
- 🔬 Then, added a bunch of new Puck/GA events to enable more granular funnel analysis on our authentication forms.

**Update**
- (https://github.com/DoSomething/northstar/pull/834/commits/a44330471f5097651a20dacf28996a46fbb65209) added a new `trackEvent` helper to track both Puck and GA events using the correct standard 
- (https://github.com/DoSomething/northstar/pull/834/commits/dcd6a5b0d23db17d8f88148363b8881aa6fc1b5e) swapped all the events to use this new helper method.

#### How should this be reviewed?
The full list of new events is in a spreadsheet linked from the Pivotal Card. 
You can find these new Puck events in the [staging Keen.io](https://keen.io/projects/59b82b6dc9e77c00014a18cb/streams) streams

Please share any feelings/suggestions on the event naming used!

#### Relevant Tickets
https://www.pivotaltracker.com/story/show/163228466

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

File: resources/assets/js/utilities/Analytics.js [hunk 1]
```diff
@@ -1,40 +1,136 @@
...
 
   Validation.Events.subscribe('Validation:Submitted', (topic, args) => {
+    // Tracks when an inline validation error free submission is made.
     Analytics.analyze('Form', 'Submitted', args);
+    puck.trackEvent('northstar_submitted_register');
   });
 
   Validation.Events.subscribe('Validation:SubmitError', (topic, args) => {
+    // Tracks when a submission is prevented due to inline validation errors.
     Analytics.analyze('Form', 'Validation Error on submit', args);
```
[the remark this answers is not in the document]
Comment (mendelB) [reply resources/assets/js/utilities/Analytics.js]: Yes! (As discussed IRL.)

(https://github.com/DoSomething/northstar/pull/834/commits/a44330471f5097651a20dacf28996a46fbb65209) added a new `trackEvent` helper to track both Puck and GA events using the correct standard and (https://github.com/DoSomething/northstar/pull/834/commits/dcd6a5b0d23db17d8f88148363b8881aa6fc1b5e) swapped all the events to use this new helper method.

File: resources/assets/js/utilities/Analytics.js [hunk 2]
```diff
@@ -1,40 +1,185 @@
...
+    trackEvent(puck, {
+      verb: 'triggered',
+      noun: 'error',
+      adjective: `field_${args}`,
+    });
   });
 
   Validation.Events.subscribe('Validation:Suggestion', (topic, args) => {
+    // Tracks email fix suggestion.
     Analytics.analyze('Form', 'Suggestion', args);
```
Comment (mendelB) [inline resources/assets/js/utilities/Analytics.js]: Should we just go ahead and add a puck event for this, and update the event name formatting?

File: resources/assets/js/utilities/Analytics.js [hunk 3]
```diff
@@ -1,40 +1,185 @@
...
   });
 
   Validation.Events.subscribe('Validation:SuggestionUsed', (topic, args) => {
+    // Tracks when email fix suggestion is used.
     Analytics.analyze('Form', 'Suggestion Used', args);
```
Comment (mendelB) [inline resources/assets/js/utilities/Analytics.js]: Same Q as above ^

File: resources/assets/js/utilities/Analytics.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/js/utilities/Analytics.js]: I think that's a good idea!

File: resources/assets/js/utilities/Analytics.js (hunk 3 above)
Comment (DFurnes) [reply resources/assets/js/utilities/Analytics.js]: Same! It'd be interesting to know how often these suggestions actually help folks out.

**Comment by mendelB** [comment]
@DFurnes Updated the suggestion events with https://github.com/DoSomething/northstar/pull/834/commits/c7a72c0a0da800a24866f9c64e67ec6139b270ce

**Pull request merged**